### PR TITLE
ci/github: Further workflow improvements

### PR DESCRIPTION
**Pull request by phlax**
- Separate verify to a stage file
- Fix disassociation of check status in PRs (hopefully)
- Attempt to restrict access based on `[bot]` trigger

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .github/workflows/_env.yml [hunk 1]
```diff
@@ -10,6 +10,12 @@ on:
         type: boolean
         default: false
 
+      trusted_bots:
+        type: string
+        default: |
+          trigger-release-envoy[bot]
+          some-other-release-envoy[bot]
```
Comment (phlax) [inline .github/workflows/_env.yml:17]: fix

File: .github/workflows/workflow-complete.yml [hunk 2]
```diff
@@ -55,7 +55,7 @@ jobs:
       uses: envoyproxy/toolshed/gh-actions/status@actions-v0.0.7
       with:
         authToken: ${{ secrets.GITHUB_TOKEN }}
-        context: ${{ github.event.workflow.name }}
+        context: Verify/examples
```
Comment (adisuissa) [inline .github/workflows/workflow-complete.yml:58]: Why is `Verify/examples` hardcoded here? shouldn't it be fetched from the github workflow name?

**Review by adisuissa — COMMENTED** [review]
Thanks for fixing this issue.
Left one question before approving.

File: .github/workflows/workflow-complete.yml (hunk 2 above)
Comment (phlax) [reply .github/workflows/workflow-complete.yml:58]: that was ~the existing problem

the workflow name is now `Publish & verify` which is unhelpful when you want both publish and verify jobs with separate status

this is a temp workaround to restore previous behaviour until i figure something better for adding additonal jobs

**Comment by phlax** [comment]
pushing through as ci is not testing the changes, and to get past coverage issue

**Pull request merged**